src/ProjectService.Business/Commands/Image/CreateImageCommand.cs

```diff
@@ -0,0 +1,120 @@
+        private readonly IRequestClient<ICreateImageRequest> _rcImages;
+        private readonly ILogger<CreateImageCommand> _logger;
+        private readonly IAccessValidator _accessValidator;
+        private readonly IHttpContextAccessor _httpContextAccessor;
+
+        private List<Guid> CreateImage(List<CreateImageData> images, List<string> errors)
+        {
+            if (images == null || images.Count == 0)
+            {
+                return null;
+            }
+
+            string errorMessage = "Can not get images. Please try again later.";
+            const string logMessage = "Errors while creating images.";
+
+            try
+            {
+                Response<IOperationResult<ICreateImagesResponse>> brokerResponse = _rcImages.GetResponse<IOperationResult<ICreateImagesResponse>>(
+                   ICreateImageRequest.CreateObj(images, ImageSource.Project)).Result;
+
+                if (brokerResponse.Message.IsSuccess && brokerResponse.Message.Body != null)
+                {
+                    return brokerResponse.Message.Body.ImagesIds;
+                }
+            }
+            catch (Exception exc)
+            {
+                _logger.LogError(exc, logMessage);
+            }
+
+            errors.Add(errorMessage);
+
+            return null;
+        }
+
+        public CreateImageCommand(
+           IImageRepository repository,
+           IRequestClient<ICreateImageRequest> rcImages,
+           ILogger<CreateImageCommand> logger,
+           IAccessValidator accessValidator,
+           IHttpContextAccessor httpContextAccessor)
+        {
+            _repository = repository;
+            _rcImages = rcImages;
+            _logger = logger;
+            _accessValidator = accessValidator;
+            _httpContextAccessor = httpContextAccessor;
+        }
+
+        public OperationResultResponse<bool> Execute(CreateImageRequest request)
+        {
+            OperationResultResponse<bool> response = new();
+
+            if (!(_accessValidator.IsAdmin() || _accessValidator.HasRights(Rights.AddEditRemoveProjects)))
+            {
+                _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
+                response.Status = OperationResultStatusType.Failed;
+                response.Errors.Add("Not enough rights.");
+
+                return response;
+            }
+
+            List<CreateImageData> images = new List<CreateImageData>
+            {
+                new CreateImageData(request.Name, request.Content, request.Extension, _httpContextAccessor.HttpContext.GetUserId())
+            };
+
+            List<Guid> imagesIds = null;
+            if (request.Image == (int)ImageType.Project)
+            {
+                imagesIds = CreateImage(images, response.Errors);
+            }
+
+            if (response.Errors.Any())
+            {
+                response.Status = OperationResultStatusType.Failed;
```

Comment:
only one way to get errors here - if something is wrong when you add images. It is not falling for creating a project.  No need return here